pkg/graph/resolvers/system_intake_relation.go

```diff
@@ -113,3 +114,40 @@ func SetSystemIntakeRelationExistingSystem(
 		return updatedIntake, nil
 	})
 }
+
+// UnlinkSystemIntakeRelation clears all the relationship information on a system intake.
+// This includes clearing the system relation type, contract name, contract number relationships, and CEDAR system relationships (TODO).
+func UnlinkSystemIntakeRelation(
+	ctx context.Context,
+	store *storage.Store,
+	intakeID uuid.UUID,
+) (*models.SystemIntake, error) {
+	return sqlutils.WithTransaction[models.SystemIntake](store, func(tx *sqlx.Tx) (*models.SystemIntake, error) {
+		// Fetch intake by ID
+		intake, err := store.FetchSystemIntakeByIDNP(ctx, tx, intakeID)
+		if err != nil {
+			return nil, err
+		}
+
+		// Clear system relation type by setting to nil
+		intake.SystemRelationType = nil
+
+		// Clear contract name
+		intake.ContractName = zero.StringFromPtr(nil)
+
+		// Clear contract number relationships by setting an empty array of contract #'s
+		if err = store.SetSystemIntakeContractNumbers(ctx, tx, intakeID, []string{}); err != nil {
```

Comment:
the only thing I'd be curious about is if `nil` would work here instead of `[]string{}`?

Reply:
There seems to be some unique handling of `nil` (compared to an empty slice) in pq.StringArray that prevents `nil` from working here. We can dig in more in the future, but I'll just resolve this as-is!